public/components/common/form/LinkInsertModal.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect, useRef } from "react"
+
+interface LinkInsertModalProps {
+  isOpen: boolean
+  onClose: () => void
+  onInsertLink: (text: string, url: string) => void
+  selectedText?: string
+}
+
+const LinkInsertModal = ({ isOpen, onClose, onInsertLink, selectedText = "" }: LinkInsertModalProps) => {
+  const [text, setText] = useState("")
+  const [url, setUrl] = useState("")
+  const [errors, setErrors] = useState<{ text?: string; url?: string }>({})
+  const textInputRef = useRef<HTMLInputElement>(null)
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault()
+
+    // Clear previous errors
+    setErrors({})
+
+    // Custom validation
+    const newErrors: { text?: string; url?: string } = {}
+
+    if (!text.trim()) {
+      newErrors.text = "Text is required"
+    }
```

Comment:
I spent a lot of time trying to get this to work with regular HTML5 validation, but because this is a modal that is triggered inside another form, I was having trouble using a regular `<button type="submit">` without `onClick={handleSubmit()}` because it would submit the create post form as well.